components/form/Form.tsx

```diff
@@ -71,32 +71,38 @@ export type ValidationRule = {
   /** validate from a regular expression */
   pattern?: RegExp;
   /** transform a value before validation */
-  transform?: (value: any) => any;
+  transform?: (value: unknown) => unknown;
```

Comment:
Not just replacing `any` with `unknown`, we also need to assert or narrow to a more specific type.